plugins/base/build.gradle.kts

```diff
@@ -33,45 +34,40 @@ dependencies {
     testImplementation(projects.core.testApi)
     testImplementation(platform(libs.junit.bom))
     testImplementation(libs.junit.jupiter)
+
+    dokkaBaseFrontendFiles(projects.plugins.base.frontend) {
+        because("fetch frontend files from subproject :plugins:base:frontend")
+    }
 }
 
-val projectDistDir = project(":plugins:base:frontend").file("dist")
-val generateFrontendFiles = tasks.getByPath(":plugins:base:frontend:generateFrontendFiles")
+// access the frontend files via the dependency on :plugins:base:frontend
+val dokkaBaseFrontendFiles: Provider<FileCollection> =
+    configurations.dokkaBaseFrontendFiles.map { frontendFiles ->
+        frontendFiles.incoming.artifacts.artifactFiles
+    }
+
+val prepareDokkaBaseFrontendFiles by tasks.registering(Sync::class) {
+    description = "copy Dokka Base frontend files into the resources directory"
 
-val copyJsFiles by tasks.registering(Copy::class) {
-    from(projectDistDir) {
+    from(dokkaBaseFrontendFiles) {
         include("*.js")
+        into("dokka/scripts")
     }
-    dependsOn(generateFrontendFiles)
-    destinationDir =
-        File(sourceSets.main.get().resources.sourceDirectories.singleFile, "dokka/scripts")
-}
 
-val copyCssFiles by tasks.registering(Copy::class) {
-    from(projectDistDir) {
+    from(dokkaBaseFrontendFiles) {
         include("*.css")
+        into("dokka/styles")
     }
-    dependsOn(generateFrontendFiles)
-    destinationDir =
-        File(sourceSets.main.get().resources.sourceDirectories.singleFile, "dokka/styles")
-}
 
-val copyFrontend by tasks.registering {
-    dependsOn(copyJsFiles, copyCssFiles)
+    into(layout.buildDirectory.dir("generated/src/main/resources"))
 }
```

Comment:
Currently the frontend files are copied directly into `src/main/resources`. This works fine, but it might cause false-positives if there's an issue where the frontend files aren't generated - any previous files will remain and continue to work, until you run `clean`.

[Sync tasks are better for mirroring files](https://docs.gradle.org/current/userguide/working_with_files.html#sec:sync_task). However, they will delete any other files in the target directory...

So instead I synced the files into a new directory, `build/generated/src/main/...` and converted the task to be a file provider, and added that file to the resources dir.

Which IntelliJ also recognises

<img width="357" alt="image" src="https://user-images.githubusercontent.com/897017/232855442-e4c9b231-9ded-4692-bfbd-e7da004a708f.png">